app/ide-desktop/lib/client/src/index.ts

```diff
@@ -47,16 +72,57 @@ class App {
     isQuitting = false
 
     async run() {
-        const { args, windowSize, chromeOptions } = configParser.parseArgs()
-        this.args = args
+        // On macOS when Enso-associated file is opened, the application is first started and then it
+        // receives the `open-file` event. However, if there is already an instance of Enso running,
+        // it receives the `open-file` event (and no new instance is created for us). In this case,
+        // we manually start a new instance of the application and pass the file path to it (using the
+        // Windows-style command).
+        electron.app.on('open-file', (event, path) => {
+            const extension = pathModule.extname(path).toLowerCase()
+            if (
+                extension === `.${paths.BUNDLED_PROJECT_EXTENSION}` ||
+                extension === `.${paths.SOURCE_FILE_EXTENSION}`
+            ) {
+                // Not the file extension we care about, let the default handler handle it.
+                return
+            }
+
+            // If we are not ready, we can still decide to open a project rather than enter the welcome
+            // screen. However, we still check for the presence of arguments, to prevent hijacking the
+            // user-spawned IDE instance (OS-spawned will not have arguments set).
+            if (!electron.app.isReady() && paths.clientArguments.length === 0) {
+                event.preventDefault()
+                console.log(`Opening file '${path}'.`)
+                APP.handleOpenFile(path)
+            } else {
+                // We need to start another copy of the application, as the first one is already running.
+                console.log(`We are already initialized, opening '${path}' in a new instance.`)
+                const args = [path]
+                const child = child_process.spawn(process.execPath, args, {
+                    detached: true,
+                    stdio: 'ignore',
+                })
+                child.unref()
```

Comment:
what it does? I think its so unpopular that a short comment in the code would be amazing.